### ISSUE TITLE
Option / flag name brainstorming

### ISSUE THREAD

**Issue by d5000** [slimcoin-project/pacli#7]
I'm going through the options and flags to try to find some with unique characters at the start to allow for most (at least for the most popular ones) an unique one-letter variant. (AFAIK, as we have already discussed, the one-character variant has always to use the first letter of the complete flag/option/variable name).

In this thread I'll present some commands where I still didn't find an appropiate one-letter flag. Those starting with the letter h should be avoided as h always should be available for --help.

----

The first "group" of flags which I perhaps need some ideas/brainstorming is the following: Most transaction commands for dPoD tokens use the `--confirm` flag to wait for a single confirmation (with a "spinner" animation). This however conflicts with the `--change` flag for a change address which is also present in almost all transaction-creating commands. I think "change address" is too established to change it.

An alternative would be --wait-for-confirmation or so, but --wait is another flag: it can be used to submit a signalling/locking/donation transaction in the next round it would fit (I guess you remember there's signalling/locking and donation rounds).

I've thought about changing `--wait` to `--presubmit` or so, but the letter "p" is also used many times, especially for `--proposal`.

**Comment by buhtignew** [comment]
Here are some ideas:
`--validate`
`--ratify`
`--defer`
`--delay`
`--stay`
`--retain`
`--temporize`
`--tarry`
If nothing of these works for you I can think of more.

**Comment by d5000** [comment]
Thanks! `--retain `looks good but unfortunately I found that it conflicts with `--reserve` which is also an important option for many donation commands ... :( The same for `--defer`.  I forgot to mention it but it conflicts with `--debug`.

`--temporize` could be ok, while there are some options/flags with "t" they are few and they could be replaced. I've not heard the word many times, but if it's used in a way similar to "wait" or "hold" then it could work :)

Another idea: `--wait` could become `--match_round` (in the sense of "match the next round where you can signal/donate/lock/vote"). `-m` would be ideal as it's not used in these commands at all.

I'll continue to brainstorm a bit, you can of course continue to post ideas ...

**Comment by buhtignew** [comment]
Is it possible I haven't understood your question?
I thought we were looking for the flag to let people waiting for the 1 confirmation when they make the transaction to know that the transaction went through.
If so I'm not sure how is it related to "matching the round". Do you mean that after one confirmation one can do other transactions of the same kind?

I think the first meaning of temporize is "delay" so it's basically the same as "wait".

**Comment by d5000** [comment]
These are two different flags, `--confirm` and `--wait` (I would like to replace one of both). `--confirm` indeed waits for a confirmation and does not wait for a specific round to "arrive" (i.e. wait for a specific block height). However `--wait` does exactly this: it waits for the next round where a specific kind of transaction (signalling/locking/release/voting) transaction can be submitted, according to the token rules. The reason why I included "wait" here in this complex is that something based on the word "wait" / "wait for confirmation" would be a good replacement for "confirm" in my opinion.

So basically this idea would be:
- replace `--confirm` with `--wait` or `--wait-for-confirmation` (this would be `-w` then).
- replace `--wait` with `--match-round` or so (`-m`).

The other idea is leaving `--wait` as it is now, and instead replacing `--confirm` by `--temporize`.

The question is which version is clearer for the user, but I think one of both could work.

Sorry to have caused confusion :)

**Comment by buhtignew** [comment]
I think `--wait-for-confirmation` and `--match-round` are much clearer then `--temporize` in each case.

As alternative we can replace both with `--temporize` but we'll have to put an effort to make its usage clear for both situations which is of course not the ideal.

**Comment by buhtignew** [comment]
Should we close this one?

**Comment by d5000** [comment]
I think this one can stay open. I have still to think about some of the flags, above all on big commands like `transaction list` I would like to have more short one-letter forms. Then I still haven't changed the flags in the dPoD token commands I listed before.

Also, this thread can be used any time you, or me, find a flag or option name which could be improved or made more intiuitive.

**Comment by d5000** [comment]
Flag names are almost done, so this topic can be soon closed, with one exception:

`donation lock` and `donation release` have the options `--reserve` and `--reserveamount`. These are the amounts which can be reserved for the next round in the case the slot was filled.

This is quite challenging. I'm searching for two flag names which both doesn't start neither with letter "R", nor with the letters:

H, D, W, T, C, N, M, S, L, Q, V (important)
A, F (less important)

Possible letters are thus B, E, G, I, J, K, O, P, U, X, Y, Z

I'll of course also brainstorm here, but have still nothing really convincing.

**Comment by buhtignew** [comment]
I think `--pick-reserveaddress` and `--include-reserveamount` are fine.

I haven't begun testing that part yet, maybe when I'm more familiar with `donation lock` and `donation release` commands I will be able to contribute better, but as now those flags sound good to me.

**Comment by d5000** [comment]
This entry is not about flags, but about whole commands: I shortened two commands.

`podtoken deck_spawn` is now `podtoken spawn`

`podtoken deck_state` is now `podtoken state`

Both don't need the "deck" in it, and this way the "deck" keyword is also a bit less present for users who prefer "token". Commit 5cd2378.

**Comment by buhtignew** [comment]
> This entry is not about flags, but about whole commands: I shortened two commands.

Maybe we could edit the name of this topic into "Flag's and command's names brainstorming", how does it sound to you? 

> I shortened two commands.

Taken note.

**Comment by buhtignew** [comment]
Would it make sense to change the `-e` flag for the `pobtoken spawn` and `podtoken spawn` commands into `-t`, i.e. `--to_block` or `--till_block`? Then it would be more in line with the meaning of the corresponding `--from_block` flag. Or we can rename the `--from_block` flag into `--first_block`.

**Comment by d5000** [comment]
-t will conflict with the token_name variable. While this could be changed, the reason I used -e is to use the same scheme than the transaction list command.
